app/connectors/FinancialDetailsConnector.scala

```diff
@@ -243,33 +241,28 @@ class FinancialDetailsConnector @Inject()(
       .map { response =>
         response.status match {
           case OK =>
-            Logger("application").debug(s"Status: ${response.status}, json: ${response.json}")
-            response.json.validate[OutstandingChargesModel].fold(
+            Logger("application").info(s"Status: ${response.status}, json: ${response.json}")
+            response.json.validate[Seq[Payment]].fold(
               invalid => {
-                Logger("application").error(s"Json Validation Error: $invalid")
-                OutstandingChargesErrorModel(Status.INTERNAL_SERVER_ERROR, "Json Validation Error. Parsing OutstandingCharges Data Response")
+                Logger("application").error(s"Json validation error: $invalid")
+                PaymentsError(response.status, "Json validation error")
               },
-              valid => valid
+              valid => Payments(valid)
             )
           case status if status >= 500 =>
             Logger("application").error(s"Status: ${response.status}, body: ${response.body}")
-            OutstandingChargesErrorModel(response.status, response.body)
-          case _ =>
-            Logger("application").warn(s"Status: ${response.status}, body: ${response.body}")
-            OutstandingChargesErrorModel(response.status, response.body)
+            PaymentsError(status, response.body)
+          case status =>
+            Logger("application").warn(s"Status ${response.status}, body: ${response.body}")
+            PaymentsError(status, response.body)
         }
-      }.recover {
-        case ex =>
-          Logger("application").error(s"Unexpected failure, ${ex.getMessage}", ex)
-          OutstandingChargesErrorModel(Status.INTERNAL_SERVER_ERROR, s"Unexpected failure, ${ex.getMessage}")
       }
   }
 
-  def getPayments(taxYear: Int)
+  def getPayments(taxYearFrom: TaxYear, taxYearTo: TaxYear)
                  (implicit headerCarrier: HeaderCarrier, mtdUser: MtdItUser[_]): Future[PaymentsResponse] = {
-
-    val dateFrom: String = s"${taxYear - 1}-04-06"
-    val dateTo: String = s"$taxYear-04-05"
+    val dateFrom: String = taxYearFrom.toFinancialYearStart.format(DateTimeFormatter.ISO_DATE)
```

Comment:
:seedling: - we can incorporate required / ISO_DATE format within TaxYear type, as we duplicating these lines all over the code base;